### PR TITLE
[ENG-343] connect to search api 2023

### PR DESCRIPTION
**Pull request by mantagen**
## Description

- update search hooks to fetch from 2020/2023 search api depending on viewType
- refactors some of the search code (separating: schema -> types -> helpers -> hooks -> components)
- adds search 2023 endpoint to config

## How to test

- run OWA (this branch) [or check preview deploy]
- go to /teachers-2023/search
- perform a search
- should see 2023 search results
- clicking search result should take you to appropriate lesson or unit page

### CODE REVIEW

File: src/context/Search/search-api/2023/constructElasticQuery.test.ts [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+
+import { constructElasticQuery } from "./constructElasticQuery";
+
+describe("Search/2023/constructElasticQuery", () => {
+  test("handles search term (without key stages)", () => {
+    const elasticQuery = constructElasticQuery(
+      createSearchQuery({ term: "writing" })
+    );
+
+    console.log(JSON.stringify(elasticQuery));
```
Comment (CraigMcCahill) [inline src/context/Search/search-api/2023/constructElasticQuery.test.ts:11]: Remove this perhaps?

**Comment by jimCresswell** [comment]
This runs against a local instance of Elasticsearch right? And there will need to be a separate piece of work to add the new endpoint URL to the config and to pull that in to OWA?

**Review by jimCresswell — APPROVED** [review]

**Review by jack-skerrett-oak — APPROVED** [review]
Some outstanding "things", spoke to Matt, happy to approve this for now!

**Pull request merged**